### ISSUE TITLE
Install not working on OS X El Capitan

### ISSUE THREAD

**Issue by hypery2k** [phonegap/ios-deploy#161]
Cannot use ios-deploy on el capitan (OS X 10.11):

```bash
$ npm -g install iso-deploy
```
I'm getting the following error:
```
|
> ios-deploy@1.7.0 preinstall /usr/local/lib/node_modules/ios-deploy
> make ios-deploy

gcc -ObjC -g -o ios-deploy -framework Foundation -framework CoreFoundation -framework MobileDevice -F/System/Library/PrivateFrameworks ios-deploy.c
In file included from ios-deploy.c:4:
In file included from /System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:10:
In file included from /System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:6:
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:7:15: error: expected ';' after @class
@class NSArray<ObjectType>;
              ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:7:16: error: cannot find protocol declaration for 'ObjectType'
@class NSArray<ObjectType>;
               ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:34:25: error: cannot find protocol declaration for 'ObjectType'
@interface NSEnumerator<ObjectType> : NSObject <NSFastEnumeration>
                        ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:34:37: error: expected identifier or '('
@interface NSEnumerator<ObjectType> : NSObject <NSFastEnumeration>
                                    ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:36:13: error: expected ')'
- (nullable ObjectType)nextObject;
            ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:36:3: note: to match this '('
- (nullable ObjectType)nextObject;
  ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:40:25: error: cannot find protocol declaration for 'ObjectType'
@interface NSEnumerator<ObjectType> (NSExtendedEnumerator)
                        ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:40:1: error: duplicate interface definition for class 'NSEnumerator'
@interface NSEnumerator<ObjectType> (NSExtendedEnumerator)
^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:34:12: note: previous definition is here
@interface NSEnumerator<ObjectType> : NSObject <NSFastEnumeration>
           ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:40:37: error: method type specifier must start with '-' or '+'
@interface NSEnumerator<ObjectType> (NSExtendedEnumerator)
                                    ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:40:38: error: expected a type
@interface NSEnumerator<ObjectType> (NSExtendedEnumerator)
                                     ^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:42:1: error: expected selector for Objective-C method
@property (readonly, copy) NSArray<ObjectType> *allObjects;
^
/System/Library/Frameworks/Foundation.framework/Headers/NSEnumerator.h:42:36: error: cannot find protocol declaration for 'ObjectType'
@property (readonly, copy) NSArray<ObjectType> *allObjects;
                                   ^
In file included from ios-deploy.c:4:
In file included from /System/Library/Frameworks/Foundation.framework/Headers/Foundation.h:10:
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:16:32: error: cannot find protocol declaration for 'ObjectType'
@interface NSArray<__covariant ObjectType> : NSObject <NSCopying, NSMutableCopying, NSSecureCoding, NSFastEnumeration>
                               ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:16:44: error: expected identifier or '('
@interface NSArray<__covariant ObjectType> : NSObject <NSCopying, NSMutableCopying, NSSecureCoding, NSFastEnumeration>
                                           ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:19:4: error: expected a type
- (ObjectType)objectAtIndex:(NSUInteger)index;
   ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:21:40: error: expected ')'
- (instancetype)initWithObjects:(const ObjectType [])objects count:(NSUInteger)cnt NS_DESIGNATED_INITIALIZER;
                                       ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:21:33: note: to match this '('
- (instancetype)initWithObjects:(const ObjectType [])objects count:(NSUInteger)cnt NS_DESIGNATED_INITIALIZER;
                                ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:26:20: error: cannot find protocol declaration for 'ObjectType'
@interface NSArray<ObjectType> (NSExtendedArray)
                   ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:26:1: error: duplicate interface definition for class 'NSArray'
@interface NSArray<ObjectType> (NSExtendedArray)
^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:16:12: note: previous definition is here
@interface NSArray<__covariant ObjectType> : NSObject <NSCopying, NSMutableCopying, NSSecureCoding, NSFastEnumeration>
           ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:26:32: error: method type specifier must start with '-' or '+'
@interface NSArray<ObjectType> (NSExtendedArray)
                               ^
/System/Library/Frameworks/Foundation.framework/Headers/NSArray.h:26:33: error: expected a type
@interface NSArray<ObjectType> (NSExtendedArray)
                                ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [ios-deploy] Error 1
npm ERR! Darwin 15.0.0
npm ERR! argv "node" "/usr/local/bin/npm" "-g" "install" "ios-deploy"
npm ERR! node v0.12.7
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! ios-deploy@1.7.0 preinstall: `make ios-deploy`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the ios-deploy@1.7.0 preinstall script 'make ios-deploy'.
npm ERR! This is most likely a problem with the ios-deploy package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     make ios-deploy
npm ERR! You can get their info via:
npm ERR!     npm owner ls ios-deploy
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /private/tmp/npm-debug.log
```

debug-log:
```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', '-g', 'install', 'ios-deploy' ]
2 info using npm@2.13.3
3 info using node@v0.12.7
4 verbose install initial load of /usr/local/lib/package.json
5 verbose readDependencies loading dependencies from /usr/local/lib/package.json
6 silly cache add args [ 'ios-deploy', null ]
7 verbose cache add spec ios-deploy
8 silly cache add parsed spec { raw: 'ios-deploy',
8 silly cache add   scope: null,
8 silly cache add   name: 'ios-deploy',
8 silly cache add   rawSpec: '',
8 silly cache add   spec: '*',
8 silly cache add   type: 'range' }
9 silly addNamed ios-deploy@*
10 verbose addNamed "*" is a valid semver range for ios-deploy
11 silly addNameRange { name: 'ios-deploy', range: '*', hasData: false }
12 silly mapToRegistry name ios-deploy
13 silly mapToRegistry using default registry
14 silly mapToRegistry registry https://registry.npmjs.org/
15 silly mapToRegistry uri https://registry.npmjs.org/ios-deploy
16 verbose addNameRange registry:https://registry.npmjs.org/ios-deploy not in flight; fetching
17 verbose request uri https://registry.npmjs.org/ios-deploy
18 verbose request no auth needed
19 info attempt registry request try #1 at 6:30:46 AM
20 verbose request id c761a0b34bdb5a80
21 verbose etag "BYAY2ZRYPBENUKZ34DYCXQ8Q0"
22 http request GET https://registry.npmjs.org/ios-deploy
23 http 304 https://registry.npmjs.org/ios-deploy
24 silly get cb [ 304,
24 silly get   { date: 'Wed, 05 Aug 2015 04:30:46 GMT',
24 silly get     via: '1.1 varnish',
24 silly get     'cache-control': 'max-age=60',
24 silly get     etag: '"BYAY2ZRYPBENUKZ34DYCXQ8Q0"',
24 silly get     age: '0',
24 silly get     connection: 'keep-alive',
24 silly get     'x-served-by': 'cache-fra1233-FRA',
24 silly get     'x-cache': 'MISS',
24 silly get     'x-cache-hits': '0',
24 silly get     'x-timer': 'S1438749046.657669,VS0,VE717',
24 silly get     vary: 'Accept' } ]
25 verbose etag https://registry.npmjs.org/ios-deploy from cache
26 verbose get saving ios-deploy to /var/root/.npm/registry.npmjs.org/ios-deploy/.cache.json
27 silly addNameRange number 2 { name: 'ios-deploy', range: '*', hasData: true }
28 silly addNameRange versions [ 'ios-deploy',
28 silly addNameRange   [ '1.0.1',
28 silly addNameRange     '1.0.2',
28 silly addNameRange     '1.0.3',
28 silly addNameRange     '1.0.4',
28 silly addNameRange     '1.0.5',
28 silly addNameRange     '1.0.6',
28 silly addNameRange     '1.0.7',
28 silly addNameRange     '1.0.8',
28 silly addNameRange     '1.0.9',
28 silly addNameRange     '1.1.0',
28 silly addNameRange     '1.2.0',
28 silly addNameRange     '1.3.0',
28 silly addNameRange     '1.3.1',
28 silly addNameRange     '1.3.2',
28 silly addNameRange     '1.4.0',
28 silly addNameRange     '1.5.0',
28 silly addNameRange     '1.6.0',
28 silly addNameRange     '1.6.1',
28 silly addNameRange     '1.6.2',
28 silly addNameRange     '1.6.3',
28 silly addNameRange     '1.7.0' ] ]
29 silly addNamed ios-deploy@1.7.0
30 verbose addNamed "1.7.0" is a plain semver version for ios-deploy
31 silly cache afterAdd ios-deploy@1.7.0
32 verbose afterAdd /var/root/.npm/ios-deploy/1.7.0/package/package.json not in flight; writing
33 verbose afterAdd /var/root/.npm/ios-deploy/1.7.0/package/package.json written
34 silly install resolved [ { name: 'ios-deploy',
34 silly install resolved     version: '1.7.0',
34 silly install resolved     os: [ 'darwin' ],
34 silly install resolved     description: 'launch iOS apps iOS devices from the command line (Xcode 6)',
34 silly install resolved     main: 'ios-deploy',
34 silly install resolved     scripts: { preinstall: 'make ios-deploy' },
34 silly install resolved     bin: { 'ios-deploy': './ios-deploy' },
34 silly install resolved     repository:
34 silly install resolved      { type: 'git',
34 silly install resolved        url: 'git+https://github.com/phonegap/ios-deploy.git' },
34 silly install resolved     keywords: [ 'ios-deploy', 'deploy to iOS device' ],
34 silly install resolved     author: { name: 'Greg Hughes' },
34 silly install resolved     license: 'GPLv3',
34 silly install resolved     gitHead: '6d61b7ca7c0ff16caea4e87a91ee7535db888621',
34 silly install resolved     bugs: { url: 'https://github.com/phonegap/ios-deploy/issues' },
34 silly install resolved     homepage: 'https://github.com/phonegap/ios-deploy#readme',
34 silly install resolved     _id: 'ios-deploy@1.7.0',
34 silly install resolved     _shasum: '8dfbce5c23ea74e77cb12870b2e72a3af8e1db91',
34 silly install resolved     _from: 'ios-deploy@*',
34 silly install resolved     _npmVersion: '2.9.1',
34 silly install resolved     _nodeVersion: '0.10.30',
34 silly install resolved     _npmUser: { name: 'shazron', email: 'shazron@gmail.com' },
34 silly install resolved     maintainers: [ [Object], [Object], [Object] ],
34 silly install resolved     dist:
34 silly install resolved      { shasum: '8dfbce5c23ea74e77cb12870b2e72a3af8e1db91',
34 silly install resolved        tarball: 'http://registry.npmjs.org/ios-deploy/-/ios-deploy-1.7.0.tgz' },
34 silly install resolved     directories: {},
34 silly install resolved     _resolved: 'https://registry.npmjs.org/ios-deploy/-/ios-deploy-1.7.0.tgz',
34 silly install resolved     readme: 'ERROR: No README data found!' } ]
35 info install ios-deploy@1.7.0 into /usr/local/lib
36 info installOne ios-deploy@1.7.0
37 verbose installOne of ios-deploy to /usr/local/lib not in flight; installing
38 verbose lock using /var/root/.npm/_locks/ios-deploy-198957223cd98201.lock for /usr/local/lib/node_modules/ios-deploy
39 silly install write writing ios-deploy 1.7.0 to /usr/local/lib/node_modules/ios-deploy
40 verbose unbuild lib/node_modules/ios-deploy
41 silly gentlyRm /usr/local/lib/node_modules/ios-deploy is being purged from base /usr/local
42 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/ios-deploy
43 verbose tar unpack /var/root/.npm/ios-deploy/1.7.0/package.tgz
44 verbose tar unpacking to /usr/local/lib/node_modules/ios-deploy
45 silly gentlyRm /usr/local/lib/node_modules/ios-deploy is being purged
46 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/ios-deploy
47 silly gunzTarPerm modes [ '755', '644' ]
48 silly gunzTarPerm extractEntry package.json
49 silly gunzTarPerm extractEntry .npmignore
50 silly gunzTarPerm extractEntry README.md
51 silly gunzTarPerm extractEntry LICENSE
52 silly gunzTarPerm extractEntry ResourceRules.plist
53 silly gunzTarPerm extractEntry Makefile
54 silly gunzTarPerm extractEntry MobileDevice.h
55 silly gunzTarPerm extractEntry CONTRIBUTING.md
56 silly gunzTarPerm extractEntry Info.plist
57 silly gunzTarPerm extractEntry demo.c
58 silly gunzTarPerm extractEntry errors.h
59 silly gunzTarPerm extractEntry ios-deploy.c
60 silly gunzTarPerm extractEntry Entitlements.plist
61 silly gunzTarPerm extractEntry resources/buildbox/build.sh
62 verbose write writing to /usr/local/lib/node_modules/ios-deploy/package.json
63 info preinstall ios-deploy@1.7.0
64 verbose unsafe-perm in lifecycle false
65 info ios-deploy@1.7.0 Failed to exec preinstall script
66 verbose unlock done using /var/root/.npm/_locks/ios-deploy-198957223cd98201.lock for /usr/local/lib/node_modules/ios-deploy
67 verbose stack Error: ios-deploy@1.7.0 preinstall: `make ios-deploy`
67 verbose stack Exit status 2
67 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:214:16)
67 verbose stack     at EventEmitter.emit (events.js:110:17)
67 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:24:14)
67 verbose stack     at ChildProcess.emit (events.js:110:17)
67 verbose stack     at maybeClose (child_process.js:1015:16)
67 verbose stack     at Process.ChildProcess._handle.onexit (child_process.js:1087:5)
68 verbose pkgid ios-deploy@1.7.0
69 verbose cwd /private/tmp
70 error Darwin 15.0.0
71 error argv "node" "/usr/local/bin/npm" "-g" "install" "ios-deploy"
72 error node v0.12.7
73 error npm  v2.13.3
74 error code ELIFECYCLE
75 error ios-deploy@1.7.0 preinstall: `make ios-deploy`
75 error Exit status 2
76 error Failed at the ios-deploy@1.7.0 preinstall script 'make ios-deploy'.
76 error This is most likely a problem with the ios-deploy package,
76 error not with npm itself.
76 error Tell the author that this fails on your system:
76 error     make ios-deploy
76 error You can get their info via:
76 error     npm owner ls ios-deploy
76 error There is likely additional logging output above.
77 verbose exit [ 1, true ]
78 verbose unbuild lib/node_modules/ios-deploy
79 info preuninstall ios-deploy@1.7.0
80 info uninstall ios-deploy@1.7.0
81 verbose unbuild rmStuff ios-deploy@1.7.0 from /usr/local/lib/node_modules
82 silly gentlyRm /usr/local/bin/ios-deploy is being gently removed
83 silly gentlyRm verifying /usr/local is an npm working directory
84 silly gentlyRm containing path /usr/local is under npm's control, in /usr/local
85 silly gentlyRm deletion target /usr/local/bin/ios-deploy is under /usr/local
86 verbose gentlyRm vacuuming from /usr/local/bin/ios-deploy up to /usr/local
87 info postuninstall ios-deploy@1.7.0
88 silly gentlyRm /usr/local/lib/node_modules/ios-deploy is being purged from base /usr/local
89 verbose gentlyRm don't care about contents; nuking /usr/local/lib/node_modules/ios-deploy
90 silly vacuum-fs purging /usr/local/lib/node_modules/ios-deploy
91 silly vacuum-fs quitting because other entries in /usr/local/lib/node_modules
```

**Comment by obetomuniz** [comment]
Any solution?

**Comment by obetomuniz** [comment]
I fix it changing my `xcode-select -print-path` to use XCode-beta.app, but now `cordova build/run ios` don't work and show me this error:

```ssh
CompileXIB test-build-ionic/Classes/MainViewController.xib
    cd /Users/obetomuniz/Workspace/lab/test-build-ionic/platforms/ios
    export PATH="/Applications/Xcode-beta.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode-beta.app/Contents/Developer/usr/bin:/Users/obetomuniz/dotfiles/apps:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/obetomuniz/.rvm/bin"
    export XCODE_DEVELOPER_USR_PATH=/Applications/Xcode-beta.app/Contents/Developer/usr/bin/..
    /Applications/Xcode-beta.app/Contents/Developer/usr/bin/ibtool --errors --warnings --notices --module test_build_ionic --output-partial-info-plist /Users/obetomuniz/Workspace/lab/test-build-ionic/platforms/ios/build/test-build-ionic.build/Debug-iphonesimulator/test-build-ionic.build/MainViewController-PartialInfo.plist --auto-activate-custom-fonts --target-device iphone --target-device ipad --minimum-deployment-target 6.0 --output-format human-readable-text --compile /Users/obetomuniz/Workspace/lab/test-build-ionic/platforms/ios/build/emulator/test-build-ionic.app/MainViewController.nib /Users/obetomuniz/Workspace/lab/test-build-ionic/platforms/ios/test-build-ionic/Classes/MainViewController.xib
2015-08-06 13:00:11.606 ibtoold[2788:54282]  DVTAssertions: Warning in /Library/Caches/com.apple.xbs/Sources/DVTFrameworks/DVTFrameworks-8164.3/DVTFoundation/Downloadables/DVTDownloadableIndex.m:73
Details:  Error Domain=DVTDownloadableErrors Code=-1 "Failed to download index with errors: (
    "Error Domain=DVTDownloadableErrors Code=-1 \"Failed to open 'https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex' with HTTP status 404: not found\" UserInfo={NSLocalizedDescription=Failed to open 'https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex' with HTTP status 404: not found, NSHTTPURLResponse=<NSHTTPURLResponse: 0x7ffa7b1cbf40> { URL: https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex } { status code: 404, headers {\n    \"Content-Encoding\" = gzip;\n    \"Content-Type\" = \"text/html\";\n    Date = \"Thu, 06 Aug 2015 16:00:10 GMT\";\n    Server = nginx;\n    \"Strict-Transport-Security\" = \"max-age=31536000; includeSubDomains\";\n    \"x-frame-options\" = SAMEORIGIN;\n    \"x-xss-protection\" = \"1; mode=block\";\n} }}",
    "Error Domain=NSCocoaErrorDomain Code=3840 \"Encountered unknown tag html on line 2\" UserInfo={NSDebugDescription=Encountered unknown tag html on line 2, kCFPropertyListOldStyleParsingError=Error Domain=NSCocoaErrorDomain Code=3840 \"Malformed data byte group at line 1; invalid hex\" UserInfo={NSDebugDescription=Malformed data byte group at line 1; invalid hex}}"
)" UserInfo={NSLocalizedDescription=Failed to download index with errors: (
    "Error Domain=DVTDownloadableErrors Code=-1 \"Failed to open 'https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex' with HTTP status 404: not found\" UserInfo={NSLocalizedDescription=Failed to open 'https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex' with HTTP status 404: not found, NSHTTPURLResponse=<NSHTTPURLResponse: 0x7ffa7b1cbf40> { URL: https://developer.apple.com/library/downloads/docset-index-7.0.0-AABB7188-E14E-4433-AD3B-5CD791EAD9A3.dvtdownloadableindex } { status code: 404, headers {\n    \"Content-Encoding\" = gzip;\n    \"Content-Type\" = \"text/html\";\n    Date = \"Thu, 06 Aug 2015 16:00:10 GMT\";\n    Server = nginx;\n    \"Strict-Transport-Security\" = \"max-age=31536000; includeSubDomains\";\n    \"x-frame-options\" = SAMEORIGIN;\n    \"x-xss-protection\" = \"1; mode=block\";\n} }}",
    "Error Domain=NSCocoaErrorDomain Code=3840 \"Encountered unknown tag html on line 2\" UserInfo={NSDebugDescription=Encountered unknown tag html on line 2, kCFPropertyListOldStyleParsingError=Error Domain=NSCocoaErrorDomain Code=3840 \"Malformed data byte group at line 1; invalid hex\" UserInfo={NSDebugDescription=Malformed data byte group at line 1; invalid hex}}"
)}
Object:   <DVTDownloadableIndexSourceDocs: 0x7ffa7aec3950>
Method:   -init
Thread:   <NSThread: 0x7ffa7ad8ed00>{number = 6, name = (null)}
Please file a bug at http://bugreport.apple.com with this warning message and any useful information you can provide.
```

**Comment by abdullaiev** [comment]
Same problem here.

**Comment by tjsail33** [comment]
Bumping. Running 10.11, fully updated, iojs 2.3.0, just reinstalled ionic and cordova. no dice.

**Comment by abdullaiev** [comment]
Hi, @brianleroux. I use node, updated to the latest 0.12.7 version, reinstalled ionic and cordova and then tried to install ios-deploy again. Unfortunately it didn't work out - exactly the same error. 
MacOS 10.11 (15A243d)

**Comment by tjsail33** [comment]
I've filed a bug report with apple regarding the 404. CaseID is 22215584.

**Comment by cyberwisdom** [comment]
Same here.  Updated El Capitan and xcode-beta but still getting build errors.  Reinstalled node, cordova, etc but no cigar...

**Comment by tjsail33** [comment]
it has to do with a dvtdownloadableindex file for Xcode 7.0 not being available where cordova expects it to be available i believe.

**Comment by shazron** [comment]
Not sure about that error with the cordova build thing is yet -- its strange since its just a plain Xcode project, perhaps something changed with the latest beta.

I don't think this is an El Capitan thing, but most likely an Xcode 7 beta thing. In any case, I can't test El Cap right now but I do have the latest beta. Will update once I know more.

**Comment by tjsail33** [comment]
Hi @shazron, any update? Sorry for being pushy, but this is such an important part of the build process to be missing, and any workaround while Apple gets their stuff fixed would be greatly appreciated!

**Comment by cyberwisdom** [comment]
@shazron based on the error messages I'm seeing here and on my setup, I agree that it's likely an xcode-beta 7 issue.   Any updates/workarounds would greatly be appreciated.

**Comment by obetomuniz** [comment]
The problem is the support of Cordova with XCode versions running in El Capitan.

Look this to get more informations about support of latest Cordova releases https://cordova.apache.org/#news

I downgrade my OSX to Yosemite and now all works fine.

**Comment by shazron** [comment]
On Yosemite 10.10.5, I installed ios-deploy with Xcode 7 b5 installed and set as the default environment, and it compiled, and I was able to deploy an app to a device. So from this data I think it may now be an El Cap issue and not just an Xcode 7 issue.

Unfortunately I don't have the setup here to debug this using El Cap, I can only get to this when I return back to work to the Bay Area Aug 24th :/

To all that reported but didn't confirm that they are running Xcode 7 b5, can you chime in to what your Xcode environment is?:
```
1. xcodebuild -version
2. xcode-select --print-path
```

**Comment by tjsail33** [comment]
FWIW, Apple responded to my bug report with "Engineering has determined that your bug report (22215584) is a duplicate of another issue (22146182) and will be closed."

This (to me) is an admission that its a Xcode7b5 issue on Apple's end. Maybe they'll release 7b6 soon.

**Comment by tjsail33** [comment]
FOUND A FIX:

cd /Applications/Xcode-beta.app (or wherever your Xcode 7 beta 4 is located)  
cd Contents/Developer/Platforms/iPhoneSimulator.platform  
cd Developer/SDKs/iPhoneSimulator.sdk/usr/lib  
sudo mv dyld_sim dyld_sim.orig  

Works for me on a fully updated system. Building via Ionic/Cordova is functional again.

**Comment by cyberwisdom** [comment]
Finally got it working! Thanks @tjsail33 and @shazron.  I figured it out based on both of your suggestions.

I had already installed Xcode 7 beta 5, so I went ahead and followed the fix that @tjsail33 described above.  However, I still had errors.  I figured out that my Xcode Command Line Tools were still using Xcode 6.4 by doing "xcodebuild -version" as suggested by @shazron.  I wasn't aware I needed to update  the command line tools option in Xcode Preferences.  Once I made the change, both build and emulate started working flawlessly.

SUMMARY OF MY STEPS

STEP 1 - After installing Xcode 7 beta 5, make sure the "Command Line Tools" Xcode option has the correct version selected:
![screen shot 2015-08-18 at 11 32 08 am](https://cloud.githubusercontent.com/assets/6466470/9334811/8fcb51c4-459e-11e5-893d-756397880571.png)

STEP 2 - In Terminal, check your Xcode version as suggested by @shazron by typing the following command:
xcodebuild -version

STEP 3 - Apply @tjsail33's FIX:
cd /Applications/Xcode-beta.app (or wherever your Xcode 7 beta is located)  
cd Contents/Developer/Platforms/iPhoneSimulator.platform  
cd Developer/SDKs/iPhoneSimulator.sdk/usr/lib  
sudo mv dyld_sim dyld_sim.orig  

Hope this helps someone else.

**Comment by fatty1380** [comment]
Can confirm - Works on El Cap 10.11 Beta (15A243d) + Xcode 6.4 (6E35b)

I had wiped out previous Xcode installs (6.4 and 7 beta) prior to performing the steps above.

**Comment by shazron** [comment]
Great news! 
I'll leave this issue open -- thinking of some ideas on how to mitigate or report this to the user better (i.e. reporting El Cap needs Xcode 7 and the command line tools installed correctly)

**Comment by shazron** [comment]
Closing. Feature request filed for #167

**Comment by zeorin** [comment]
I know this is closed, but I had the same issue on OS X Yosemite 10.10.1 & XCode 6.4. I installed `node` directly from the official website. My `node` version is `4.1.0` and `npm` is version `3.3.3` (after updating).

Eventually I got it to work by running `sudo npm install ios-deploy -g --unsafe-perm`.

The reason is that if `npm` is invoked as `root`, [in order to run `preinstall`, `npm` drops its elevated permissions](https://docs.npmjs.com/misc/scripts#user) (by default to the  `nobody` user), and does not have the necessary permissions to the `/usr/local/lib/node_modules/ios-deploy/ios-deploy` folder. 

I'm not an expert or anything, but the [npm script best practices](https://docs.npmjs.com/misc/scripts#best-practices) advise against using `preinstall` except for required compilation on a target architecture. I can only suppose that `ios-deploy` is such a use case...

This package is likely to be installed globally by people because they're also likely to install PhoneGap/Cordova globally. If it's not possible to work around this issue for people who need to run `npm install ios-deploy -g` with `sudo` (or people who'd like that requirement to remain in place), there ought to be some note in the installation instructions about a workaround. However, it seems to me that it's more likely a permissions issue, i.e. in order to run the `preinstall` script, the `nobody` user ought to be able to access `/usr/local/lib/node_modules/ios-deploy/ios-deploy`.
